Mathlib/Algebra/FreeMonoid/Count.lean

```diff
@@ -15,71 +14,76 @@ additive homomorphisms from `FreeMonoid` and `FreeAddMonoid`.
 
 We do not use `to_additive` because it can't map `Multiplicative ℕ` to `ℕ`.
 
-## TODO
-
-There is lots of defeq abuse here of `FreeAddMonoid α = List α`, e.g. in statements like
-```
-theorem countP_apply (l : FreeAddMonoid α) : countP p l = List.countP p l := rfl
-```
-This needs cleaning up.
-
 -/
 
 variable {α : Type*} (p : α → Prop) [DecidablePred p]
 
-namespace FreeAddMonoid
+namespace FreeMonoid
+/-- `List.countP` lifted to free monoids-/
+@[to_additive "`List.countP` lifted to free additive monoids"]
+def countP' (l:FreeMonoid α) : ℕ := l.toList.countP p
```

Comment:
```suggestion
def countP' (l : FreeMonoid α) : ℕ := l.toList.countP p
```
And can we find a name without a prime?

Reply:
the reason i added this is in order to leverage `to_additive` with the lemmas below

Reply:
i guess we could unprime this one and name the bundled versions `countPHom`? but that seems unnecessarily cumbersome